ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/WorkloadTableToolbar.tsx

```diff
@@ -1,11 +1,168 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import {
+    Toolbar,
+    ToolbarItem,
+    ToolbarFilter,
+    ToolbarToggleGroup,
+    ToolbarGroup,
+    ToolbarContent,
+} from '@patternfly/react-core';
+import { FilterIcon } from '@patternfly/react-icons';
+
+import useURLSearch from 'hooks/useURLSearch';
+import { uniq } from 'lodash';
+import { DefaultFilters } from './types';
+import FilterResourceDropdown, { Resource } from './FilterResourceDropdown';
+import FilterAutocompleteInput from './FilterAutocompleteInput';
+import CVESeverityDropdown from './CVESeverityDropdown';
+import CVEStatusDropdown from './CVEStatusDropdown';
+
+type FilterType = 'resource' | 'Severity' | 'Fixable';
+
+type WorkloadTableToolbarProps = {
+    defaultFilters: DefaultFilters;
+    resourceContext?: Resource;
+};
+
+function WorkloadTableToolbar({ defaultFilters, resourceContext }: WorkloadTableToolbarProps) {
+    const { searchFilter, setSearchFilter } = useURLSearch();
```

Comment:
I would ideally like to type check if `searchFilter` corresponds to what we expect